examples/compare_product_prices/compare_product_prices.py

```diff
@@ -0,0 +1,68 @@
+"""This example demonstrates how to compare product prices across websites with query_data() method."""
+
+from agentql.ext.playwright.sync_api import Page
+from playwright.sync_api import sync_playwright
+
+# Set the URL to the desired website
+BESTBUY_URL = "https://www.bestbuy.com/site/nintendo-switch-oled-model-w-joy-con-white/6470923.p?skuId=6470923"
+TARGET_URL = "https://www.target.com/p/nintendo-switch-oled-model-with-white-joy-con/-/A-83887639#lnk=sametab"
+NINETENDO_URL = "https://www.nintendo.com/us/store/products/nintendo-switch-oled-model-white-set/"
+
+# Define the queries to get the product price
+PRODUCT_INFO_QUERY = """
+{
+    nintendo_switch
+    {
+        price
+    }
+}"""
+
+
+def print_header():
+    """Prints the header for the data table"""
+    print(f"{'Website':<25} | {'Product ':<20} | {'Price ':<20} ")
+    print("-" * 75)
+
+
+def print_row(website, product, price):
+    """Prints the data row"""
+    print(f"{website:<25} | {product:<20} | {price:<20} ")
+
+
+def main():
+    print_header()
+
+    with sync_playwright() as playwright:
+        browser = playwright.chromium.launch(headless=False)
```

Comment:
not specific to this example, but what do you think about moving browser creation into with block as well, to automanage it as a resource?

Reply:
I think that's a good idea and will save users the pain of both creating browser and stopping browser.

But to do this, we probably need to create our own context manager API that wraps around the launch() method to make it work in both sync and async environment.

Reply:
It works out of the box